internal/exec/engine.go

```diff
@@ -184,28 +210,44 @@ func (e *Engine) acquireConfig() (cfg types.Config, err error) {
 // is unavailable. This will also render the config (see renderConfig) before
 // returning.
 func (e *Engine) fetchProviderConfig() (types.Config, error) {
-	fetchers := []providers.FuncFetchConfig{
-		cmdline.FetchConfig,
-		system.FetchConfig,
-		e.PlatformConfig.FetchFunc(),
+	fetchers := map[string]providers.FuncFetchConfig{
+		"cmdLine":  cmdline.FetchConfig,
+		"system":   system.FetchConfig,
+		"platform": e.PlatformConfig.FetchFunc(),
 	}
-
 	var cfg types.Config
 	var r report.Report
 	var err error
-	for _, fetcher := range fetchers {
+	var providerKey string
+	var configPath string
+	for key, fetcher := range fetchers {
 		cfg, r, err = fetcher(e.Fetcher)
 		if err != providers.ErrNoProvider {
 			// successful, or failed on another error
+			providerKey = key
 			break
 		}
 	}
+	if providerKey == "cmdLine" {
+		path, err := cmdline.GetCmdPath(e.Fetcher)
+		if err != nil {
+			e.Logger.Info("failed to get path for cmdLine: %v", err)
+		}
+		configPath = path
+	} else if providerKey == "system" {
+		configPath = filepath.Join(distro.SystemConfigDir(), "user.ign")
+	} else if providerKey == "platform" {
+		configPath = e.PlatformConfig.Name()
+	}
```

Comment:
Ahh OK, I see.

Hmm, how about instead of `IGNITION_FETCH_CONFIG_PATH`, we make it a more generic `IGNITION_CONFIG_SRC` instead? I.e. it's not necessarily about the exact path that was queried, but more the actual provider that was used (but for referenced configs, we still fallback to the path).

I.e. we remove all this here, use `e.PlatformConfig.Name()` in the `map[string]providers.FuncFetchConfig` above, and change the `logStructuredJournalEntry` call below to use `providerKey` instead of `configPath`.

Reply:
And if in the future we also want a more specific `PATH` type variable, maybe a cleaner approach is to enhance the fetch functions to also return the URI used and log that.